In [1]:
import pandas as pd

# Creamos dataframes de los datasets de puntos (ajustar la ruta según tus directorios)
df_puntos_eurocup = pd.read_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\Datasets\eurocup_points.csv")
df_puntos_euroleague = pd.read_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\Datasets\euroleague_points.csv")

# Creamos dataframes de los datasets de boxscore (ajustar la ruta según tus directorios)
df_boxscore_eurocup = pd.read_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\Datasets\eurocup_box_score.csv")
df_boxscore_euroleague = pd.read_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\Datasets\euroleague_box_score.csv")

C:\Users\Yago\AppData\Local\Temp\ipykernel_24756\4114587192.py:8: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  df_boxscore_eurocup = pd.read_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\Datasets\eurocup_box_score.csv")


# Preprocesado de datos

Vamos a comparar los dataframes de cada competición para ver que tengan el mismo número de partidos.

In [2]:
df_puntos_eurocup

game_point_id    game_id     game  round           phase season_code  \
0       U2007_001_004  U2007_001  SIA-TTK      1  REGULAR SEASON       U2007   
1       U2007_001_005  U2007_001  SIA-TTK      1  REGULAR SEASON       U2007   
2       U2007_001_007  U2007_001  SIA-TTK      1  REGULAR SEASON       U2007   
3       U2007_001_008  U2007_001  SIA-TTK      1  REGULAR SEASON       U2007   
4       U2007_001_010  U2007_001  SIA-TTK      1  REGULAR SEASON       U2007   
...               ...        ...      ...    ...             ...         ...   
527946  U2023_196_623  U2023_196  BOU-PRS     25          FINALS       U2023   
527947  U2023_196_626  U2023_196  BOU-PRS     25          FINALS       U2023   
527948  U2023_196_632  U2023_196  BOU-PRS     25          FINALS       U2023   
527949  U2023_196_634  U2023_196  BOU-PRS     25          FINALS       U2023   
527950  U2023_196_636  U2023_196  BOU-PRS     25          FINALS       U2023   

        number_of_play team_id player_id              player  ... coord_y  \
0                    4     SIA   P000168      KIEZA, AURIMAS  ...       2   
1                    5     SIA   P000643       OLIVERO, JOSE  ...     205   
2                    7     TTK      PBAT     WRIGHT, MICHAEL  ...      -1   
3                    8     TTK      PBAT     WRIGHT, MICHAEL  ...      -1   
4                   10     SIA      PBWG  MACEINA, GEDIMINAS  ...     520   
...                ...     ...       ...                 ...  ...     ...   
527946             623     PRS   P011232       SIMON, JUSTIN  ...      25   
527947             626     BOU   P011836        MIKE, ISIAHA  ...     652   
527948             632     BOU   P011836        MIKE, ISIAHA  ...     -50   
527949             634     PRS   P012613    JANTUNEN, MIKAEL  ...       0   
527950             636     BOU   P010449       BENITEZ, HUGO  ...     100   

       zone  fastbreak  second_chance  points_off_turnover minute  console  \
0         D        NaN            NaN                  NaN      1    09:00   
1         B        NaN            NaN                  NaN      1    09:00   
2       NaN        NaN            NaN                  NaN      1    09:00   
3       NaN        NaN            NaN                  NaN      1    09:00   
4         I        NaN            NaN                  NaN      2    08:00   
...     ...        ...            ...                  ...    ...      ...   
527946    A        0.0            0.0                  0.0     40    00:56   
527947    I        0.0            0.0                  0.0     40    00:48   
527948    H        0.0            1.0                  0.0     40    00:48   
527949    A        0.0            0.0                  0.0     40    00:26   
527950    C        0.0            0.0                  0.0     40    00:20   

        points_a  points_b            timestamp  
0              2         0           1970-01-01  
1              4         0           1970-01-01  
2              4         1           1970-01-01  
3              4         2           1970-01-01  
4              6         2           1970-01-01  
...          ...       ...                  ...  
527946        76        87  2024-04-12 19:28:50  
527947        76        87  2024-04-12 19:31:01  
527948        79        87  2024-04-12 19:31:40  
527949        79        89  2024-04-12 19:32:00  
527950        81        89  2024-04-12 19:32:12  

[527951 rows x 24 columns]

In [3]:
df_boxscore_eurocup

game_player_id    game_id     game  round           phase  \
0         U2007_187_PJEZ  U2007_187  JER-DEN      7  REGULAR SEASON   
1         U2007_187_PKMF  U2007_187  JER-DEN      7  REGULAR SEASON   
2         U2007_187_PKNU  U2007_187  JER-DEN      7  REGULAR SEASON   
3         U2007_187_PKQK  U2007_187  JER-DEN      7  REGULAR SEASON   
4      U2007_187_P000678  U2007_187  JER-DEN      7  REGULAR SEASON   
...                  ...        ...      ...    ...             ...   
87132  U2020_083_P006549  U2020_083  NTR-CAN      7  REGULAR SEASON   
87133  U2020_023_P006549  U2020_023  CAN-NTR      2  REGULAR SEASON   
87134  U2020_162_P006549  U2020_162  NTR-JOV     16          TOP 16   
87135  U2020_070_P006549  U2020_070  LJU-NTR      6  REGULAR SEASON   
87136  U2020_035_P006549  U2020_035  NTR-TRN      3  REGULAR SEASON   

      season_code player_id  is_starter  is_playing team_id  ...  \
0           U2007      PJEZ         0.0         1.0     JER  ...   
1           U2007      PKMF         1.0         1.0     JER  ...   
2           U2007      PKNU         1.0         1.0     JER  ...   
3           U2007      PKQK         1.0         1.0     JER  ...   
4           U2007   P000678         0.0         1.0     JER  ...   
...           ...       ...         ...         ...     ...  ...   
87132       U2020   P006549         1.0         1.0     NTR  ...   
87133       U2020   P006549         0.0         1.0     NTR  ...   
87134       U2020   P006549         1.0         1.0     NTR  ...   
87135       U2020   P006549         1.0         1.0     NTR  ...   
87136       U2020   P006549         1.0         1.0     NTR  ...   

      total_rebounds assists steals  turnovers  blocks_favour  blocks_against  \
0                  0       0      0          0              0               0   
1                  6       6      0          2              0               0   
2                  2       5      5          1              0               0   
3                  8       1      1          0              0               2   
4                  1       0      0          0              0               0   
...              ...     ...    ...        ...            ...             ...   
87132              0       6      1          4              0               0   
87133              2       0      1          2              0               0   
87134              1       5      1          1              0               1   
87135              5       3      0          2              0               2   
87136              2       4      2          1              0               0   

       fouls_committed  fouls_received  valuation  plus_minus  
0                    0               0          0          -2  
1                    3               1         10          -1  
2                    3               4         30          11  
3                    3               2          9          12  
4                    4               4         19          -2  
...                ...             ...        ...         ...  
87132                2               1         23           9  
87133                3               5          4         -18  
87134                3               0          5         -19  
87135                0               4         11          -8  
87136                4               3         12          -6  

[87137 rows x 32 columns]

Vemos que las últimas filas para ambos datasets son distintas. Vamos a comprobar las temporadas y si el último partido del dataframe puntos (que parece estar ordenado cronológicamente) también se encuentra en el dataframe boxscore

In [4]:
print("Seasons points eurocup: ", df_puntos_eurocup['season_code'].unique())
print("Seasons boxscore eurocup: ", df_boxscore_eurocup['season_code'].unique())
df_boxscore_eurocup[df_boxscore_eurocup['game_id'] == "U2023_196"]

Seasons points eurocup:  ['U2007' 'U2008' 'U2009' 'U2010' 'U2011' 'U2012' 'U2013' 'U2014' 'U2015'
 'U2016' 'U2017' 'U2018' 'U2019' 'U2020' 'U2021' 'U2022' 'U2023']
Seasons boxscore eurocup:  ['U2007' 'U2008' 'U2009' 'U2010' 'U2011' 'U2012' 'U2013' 'U2014' 'U2015'
 'U2016' 'U2017' 'U2018' 'U2019' 'U2020' 'U2021' 'U2022' 'U2023']


game_player_id    game_id     game  round   phase season_code  \
79066  U2023_196_P012617  U2023_196  BOU-PRS     25  FINALS       U2023   
79067  U2023_196_P012628  U2023_196  BOU-PRS     25  FINALS       U2023   
79068  U2023_196_P004525  U2023_196  BOU-PRS     25  FINALS       U2023   
79069  U2023_196_P010449  U2023_196  BOU-PRS     25  FINALS       U2023   
79070  U2023_196_P010450  U2023_196  BOU-PRS     25  FINALS       U2023   
79071  U2023_196_P010718  U2023_196  BOU-PRS     25  FINALS       U2023   
79072  U2023_196_P012618  U2023_196  BOU-PRS     25  FINALS       U2023   
79073  U2023_196_P010431  U2023_196  BOU-PRS     25  FINALS       U2023   
79074  U2023_196_P000498  U2023_196  BOU-PRS     25  FINALS       U2023   
79075  U2023_196_P011836  U2023_196  BOU-PRS     25  FINALS       U2023   
79076  U2023_196_P011837  U2023_196  BOU-PRS     25  FINALS       U2023   
79077  U2023_196_P011161  U2023_196  BOU-PRS     25  FINALS       U2023   
79078      U2023_196_BOU  U2023_196  BOU-PRS     25  FINALS       U2023   
79079  U2023_196_P012608  U2023_196  BOU-PRS     25  FINALS       U2023   
79080  U2023_196_P012609  U2023_196  BOU-PRS     25  FINALS       U2023   
79081  U2023_196_P012640  U2023_196  BOU-PRS     25  FINALS       U2023   
79082  U2023_196_P012610  U2023_196  BOU-PRS     25  FINALS       U2023   
79083  U2023_196_P005793  U2023_196  BOU-PRS     25  FINALS       U2023   
79084  U2023_196_P003481  U2023_196  BOU-PRS     25  FINALS       U2023   
79085  U2023_196_P012611  U2023_196  BOU-PRS     25  FINALS       U2023   
79086  U2023_196_P004512  U2023_196  BOU-PRS     25  FINALS       U2023   
79087  U2023_196_P011887  U2023_196  BOU-PRS     25  FINALS       U2023   
79088  U2023_196_P009556  U2023_196  BOU-PRS     25  FINALS       U2023   
79089  U2023_196_P012613  U2023_196  BOU-PRS     25  FINALS       U2023   
79090  U2023_196_P011232  U2023_196  BOU-PRS     25  FINALS       U2023   
79091      U2023_196_PRS  U2023_196  BOU-PRS     25  FINALS       U2023   

      player_id  is_starter  is_playing team_id  ... total_rebounds assists  \
79066   P012617         1.0         1.0     BOU  ...              4       8   
79067   P012628         0.0         1.0     BOU  ...              2       2   
79068   P004525         0.0         1.0     BOU  ...              1       1   
79069   P010449         0.0         1.0     BOU  ...              1       4   
79070   P010450         0.0         1.0     BOU  ...              0       0   
79071   P010718         1.0         1.0     BOU  ...              1       1   
79072   P012618         0.0         1.0     BOU  ...              3       0   
79073   P010431         1.0         1.0     BOU  ...              1       0   
79074   P000498         0.0         0.0     BOU  ...              0       0   
79075   P011836         1.0         1.0     BOU  ...              6       1   
79076   P011837         1.0         1.0     BOU  ...              5       0   
79077   P011161         0.0         1.0     BOU  ...              1       4   
79078       BOU         0.0         1.0     BOU  ...             28      21   
79079   P012608         1.0         1.0     PRS  ...              6       5   
79080   P012609         0.0         1.0     PRS  ...              4       1   
79081   P012640         0.0         1.0     PRS  ...              1       3   
79082   P012610         1.0         1.0     PRS  ...              4       1   
79083   P005793         0.0         0.0     PRS  ...              0       0   
79084   P003481         0.0         1.0     PRS  ...              2       1   
79085   P012611         1.0         1.0     PRS  ...              2       1   
79086   P004512         1.0         1.0     PRS  ...              6       1   
79087   P011887         0.0         0.0     PRS  ...              0       0   
79088   P009556         0.0         1.0     PRS  ...              3       1   
79089   P012613         1.0         1.0     PRS  ...              1       3   
79090   P

Todo en orden en la Eurocup. Vamos con la Euroliga.

In [5]:
print("Seasons points euroleague: ", df_puntos_euroleague['season_code'].unique())
print("Seasons boxscore euroleague: ", df_boxscore_euroleague['season_code'].unique())

Seasons points euroleague:  ['E2007' 'E2008' 'E2009' 'E2010' 'E2011' 'E2012' 'E2013' 'E2014' 'E2015'
 'E2016' 'E2017' 'E2018' 'E2019' 'E2020' 'E2021' 'E2022' 'E2023' 'E2024']
Seasons boxscore euroleague:  ['E2007' 'E2008' 'E2009' 'E2010' 'E2011' 'E2012' 'E2013' 'E2014' 'E2015'
 'E2016' 'E2017' 'E2018' 'E2019' 'E2020' 'E2021' 'E2022' 'E2023' 'E2024']


In [6]:
df_puntos_euroleague

game_point_id    game_id     game  round           phase season_code  \
0       E2007_001_002  E2007_001  OLY-BAS      1  REGULAR SEASON       E2007   
1       E2007_001_004  E2007_001  OLY-BAS      1  REGULAR SEASON       E2007   
2       E2007_001_008  E2007_001  OLY-BAS      1  REGULAR SEASON       E2007   
3       E2007_001_010  E2007_001  OLY-BAS      1  REGULAR SEASON       E2007   
4       E2007_001_012  E2007_001  OLY-BAS      1  REGULAR SEASON       E2007   
...               ...        ...      ...    ...             ...         ...   
642543  E2024_018_666  E2024_018  OLY-ZAL      2  REGULAR SEASON       E2024   
642544  E2024_018_678  E2024_018  OLY-ZAL      2  REGULAR SEASON       E2024   
642545  E2024_018_679  E2024_018  OLY-ZAL      2  REGULAR SEASON       E2024   
642546  E2024_018_680  E2024_018  OLY-ZAL      2  REGULAR SEASON       E2024   
642547  E2024_018_682  E2024_018  OLY-ZAL      2  REGULAR SEASON       E2024   

        number_of_play team_id player_id                player  ... coord_y  \
0                    2     BAS      PCBX       PRIGIONI, PABLO  ...     479   
1                    4     OLY      PTFV         WOODS, QYNTEL  ...     451   
2                    8     OLY      PCBV  MACIJAUSKAS, ARVYDAS  ...      17   
3                   10     BAS      PCBY       SPLITTER, TIAGO  ...      33   
4                   12     OLY   P000126         JACKSON, MARC  ...      76   
...                ...     ...       ...                   ...  ...     ...   
642543             666     ZAL   P007975    FRANCISCO, SYLVAIN  ...      -1   
642544             678     OLY   P003469       VEZENKOV, SASHA  ...      -1   
642545             679     OLY   P003469       VEZENKOV, SASHA  ...      -1   
642546             680     ZAL   P007975    FRANCISCO, SYLVAIN  ...     809   
642547             682     ZAL   P003210     LEKAVICIUS, LUKAS  ...     639   

       zone  fastbreak  second_chance  points_off_turnover minute  console  \
0         G        NaN            NaN                  NaN      1    09:00   
1         I        NaN            NaN                  NaN      1    09:00   
2         D        NaN            NaN                  NaN      1    09:00   
3         C        NaN            NaN                  NaN      2    08:00   
4         C        NaN            NaN                  NaN      2    08:00   
...     ...        ...            ...                  ...    ...      ...   
642543  NaN        0.0            0.0                  0.0     40    00:14   
642544  NaN        0.0            0.0                  0.0     40    00:14   
642545  NaN        0.0            0.0                  0.0     40    00:14   
642546    I        0.0            0.0                  0.0     40    00:05   
642547    H        0.0            0.0                  0.0     40    00:01   

        points_a  points_b            timestamp  
0              0         3           1970-01-01  
1              0         3           1970-01-01  
2              2         3           1970-01-01  
3              2         5           1970-01-01  
4              2         5           1970-01-01  
...          ...       ...                  ...  
642543        72        68  2024-10-11 20:35:27  
642544        73        68  2024-10-11 20:37:51  
642545        74        68  2024-10-11 20:38:05  
642546        74        68  2024-10-11 20:38:16  
642547        74        68  2024-10-11 20:38:20  

[642548 rows x 24 columns]

In [7]:
df_boxscore_euroleague

game_player_id    game_id     game  round           phase  \
0          E2007_191_PADS  E2007_191  BAR-ROM     17          TOP 16   
1          E2007_191_PAZU  E2007_191  BAR-ROM     17          TOP 16   
2          E2007_191_PJJF  E2007_191  BAR-ROM     17          TOP 16   
3          E2007_191_PAHW  E2007_191  BAR-ROM     17          TOP 16   
4          E2007_191_PATA  E2007_191  BAR-ROM     17          TOP 16   
...                   ...        ...      ...    ...             ...   
110762  E2023_262_P011649  E2023_262  BER-MAD     30  REGULAR SEASON   
110763  E2023_008_P011649  E2023_008  BAS-MAD      1  REGULAR SEASON   
110764  E2023_148_P011649  E2023_148  ASV-MAD     17  REGULAR SEASON   
110765  E2022_001_P011649  E2022_001  PAN-MAD      1  REGULAR SEASON   
110766  E2024_005_P011649  E2024_005  MUN-MAD      1  REGULAR SEASON   

       season_code player_id  is_starter  is_playing team_id  ...  \
0            E2007      PADS         1.0         1.0     BAR  ...   
1            E2007      PAZU         0.0         1.0     BAR  ...   
2            E2007      PJJF         1.0         1.0     BAR  ...   
3            E2007      PAHW         0.0         1.0     BAR  ...   
4            E2007      PATA         1.0         1.0     BAR  ...   
...            ...       ...         ...         ...     ...  ...   
110762       E2023   P011649         0.0         0.0     MAD  ...   
110763       E2023   P011649         0.0         0.0     MAD  ...   
110764       E2023   P011649         0.0         1.0     MAD  ...   
110765       E2022   P011649         0.0         0.0     MAD  ...   
110766       E2024   P011649         0.0         1.0     MAD  ...   

       total_rebounds assists steals  turnovers  blocks_favour  \
0                   2       0      1          0              0   
1                   5       3      2          2              0   
2                   8       0      1          2              0   
3                   1       0      0          0              2   
4                   2       3      1          4              0   
...               ...     ...    ...        ...            ...   
110762              0       0      0          0              0   
110763              0       0      0          0              0   
110764              0       0      0          0              0   
110765              0       0      0          0              0   
110766              1       0      0          1              0   

        blocks_against  fouls_committed  fouls_received  valuation  plus_minus  
0                    0                1               4         18          18  
1                    0                1               0          8          17  
2                    0                0               1         14           0  
3                    0                1               1          6          -5  
4                    0                1               4         18          12  
...                ...              ...             ...        ...         ...  
110762               0                0               0          0           0  
110763               0                0               0          0           0  
110764               0                2               0         -4           4  
110765               0                0               0          0           0  
110766               0                0               1          1           2  

[110767 rows x 32 columns]

Vamos a realizar la misma comprobación que antes, esta vez con los datos de Euroliga.

In [8]:
df_boxscore_euroleague[df_boxscore_euroleague['game_id'] == "E2024_018"]

game_player_id    game_id     game  round           phase  \
110352  E2024_018_P008855  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110353  E2024_018_P007982  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110354  E2024_018_P012878  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110355  E2024_018_P007947  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110356  E2024_018_P008811  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110357  E2024_018_P003469  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110358  E2024_018_P001255  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110359  E2024_018_P009849  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110360  E2024_018_P008829  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110361  E2024_018_P011194  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110362  E2024_018_P007853  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110363  E2024_018_P013380  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110364      E2024_018_OLY  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110365  E2024_018_P011925  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110366  E2024_018_P007975  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110367  E2024_018_P003210  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110368  E2024_018_P011983  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110369  E2024_018_P007513  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110370  E2024_018_P005504  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110371  E2024_018_P011051  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110372  E2024_018_P012622  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110373  E2024_018_P003048  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110374  E2024_018_P002676  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110375  E2024_018_P007029  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110376  E2024_018_P002329  E2024_018  OLY-ZAL      2  REGULAR SEASON   
110377      E2024_018_ZAL  E2024_018  OLY-ZAL      2  REGULAR SEASON   

       season_code player_id  is_starter  is_playing team_id  ...  \
110352       E2024   P008855         1.0         1.0     OLY  ...   
110353       E2024   P007982         0.0         1.0     OLY  ...   
110354       E2024   P012878         1.0         1.0     OLY  ...   
110355       E2024   P007947         1.0         1.0     OLY  ...   
110356       E2024   P008811         0.0         1.0     OLY  ...   
110357       E2024   P003469         1.0         1.0     OLY  ...   
110358       E2024   P001255         0.0         1.0     OLY  ...   
110359       E2024   P009849         0.0         1.0     OLY  ...   
110360       E2024   P008829         0.0         1.0     OLY  ...   
110361       E2024   P011194         0.0         1.0     OLY  ...   
110362       E2024   P007853         0.0         1.0     OLY  ...   
110363       E2024   P013380         1.0         1.0     OLY  ...   
110364       E2024       OLY         0.0         1.0     OLY  ...   
110365       E2024   P011925         1.0         1.0     ZAL  ...   
110366       E2024   P007975         0.0         1.0     ZAL  ...   
110367       E2024   P003210         0.0         1.0     ZAL  ...   
110368       E2024   P011983         1.0         1.0     ZAL  ...   
110369       E2024   P007513         1.0         1.0     ZAL  ...   
110370       E2024   P005504         1.0         1.0     ZAL  ...   
110371       E2024   P011051         0.0         1.0     ZAL  ...   
110372       E2024   P012622         0.0         1.0     ZAL  ...   
110373       E2024   P003048         0.0         1.0     ZAL  ...   
110374       E2024   P002676         1.0         1.0     ZAL  ...   
110375       E2024   P007029         0.0         1.0     ZAL  ...   
110376       E2024   P002329         0.0         1.0     ZAL  ...   
110377       E2024       ZAL         0.0         1.0     ZAL  ...   

       total_rebounds assists steals  turnovers  blocks_favour  \
110352              2       7      0          1              0   
110353              3       3      0          2              0   
1103

Todas las fechas cuadran, así que ahora vamos a eliminar las columnas que no nos interesen para nuestro análisis.
Primero, vamos a mostrar las columnas de cada dataset y elegiremos a partir de ahí

In [9]:
print("Columnas boxscore Euroleague: ", df_boxscore_euroleague.columns.tolist())
print("Columnas boxscore Eurocup: ", df_boxscore_eurocup.columns.tolist())
print("Columnas puntos Euroleague: ", df_puntos_euroleague.columns.tolist())
print("Columnas puntos Euroleague: ", df_puntos_eurocup.columns.tolist())

Columnas boxscore Euroleague:  ['game_player_id', 'game_id', 'game', 'round', 'phase', 'season_code', 'player_id', 'is_starter', 'is_playing', 'team_id', 'dorsal', 'player', 'minutes', 'points', 'two_points_made', 'two_points_attempted', 'three_points_made', 'three_points_attempted', 'free_throws_made', 'free_throws_attempted', 'offensive_rebounds', 'defensive_rebounds', 'total_rebounds', 'assists', 'steals', 'turnovers', 'blocks_favour', 'blocks_against', 'fouls_committed', 'fouls_received', 'valuation', 'plus_minus']
Columnas boxscore Eurocup:  ['game_player_id', 'game_id', 'game', 'round', 'phase', 'season_code', 'player_id', 'is_starter', 'is_playing', 'team_id', 'dorsal', 'player', 'minutes', 'points', 'two_points_made', 'two_points_attempted', 'three_points_made', 'three_points_attempted', 'free_throws_made', 'free_throws_attempted', 'offensive_rebounds', 'defensive_rebounds', 'total_rebounds', 'assists', 'steals', 'turnovers', 'blocks_favour', 'blocks_against', 'fouls_committed'

Columnas Relevantes para Boxscore (Euroleague y Eurocup)

- game_id: Identificador único del partido, esencial para unir esta información con los datos de puntos.
- team_id: Identificador del equipo, útil para agrupar y filtrar los datos por equipo.
- player_id: Identificador del jugador para hacer el seguimiento individual.
- points: Cantidad de puntos anotados por el jugador, clave para calcular si alcanza o supera los 10 puntos.

In [10]:
# Eliminamos el resto de columnas
df_puntos_euroleague = df_puntos_euroleague.drop(columns = ['game_point_id', 'game', 'minute', 'round', 'phase', 'season_code', 'number_of_play', 'player_id', 'player', 'action_id', 'action', 'coord_x', 'coord_y', 'zone', 'fastbreak', 'second_chance', 'points_off_turnover', 'console', 'timestamp'])
df_puntos_eurocup = df_puntos_eurocup.drop(columns = ['game_point_id', 'game', 'minute', 'round', 'phase', 'season_code', 'number_of_play', 'player_id', 'player', 'action_id', 'action', 'coord_x', 'coord_y', 'zone', 'fastbreak', 'second_chance', 'points_off_turnover', 'console', 'timestamp'])

Columnas Relevantes para Datos de Puntos

- game_id: Necesario para enlazar con los datos de boxscore.
- team_id: Útil para saber a qué equipo pertenece cada anotación.
- points: Esencial para sumar los puntos de cada anotación y ver cuándo cada equipo alcanza el hito de 69 puntos.
- points_a y points_b: Estas columnas pueden ser útiles para rastrear el marcador de ambos equipos y verificar en qué punto cada equipo llega a los hitos de 69 o 10 puntos.

In [11]:
# Eliminamos el resto de columnas
df_boxscore_euroleague = df_boxscore_euroleague.drop(columns = ['game_player_id', 'minutes', 'game', 'round', 'phase', 'season_code', 'is_starter', 'is_playing', 'dorsal', 'player', 'two_points_made', 'two_points_attempted', 'three_points_made', 'three_points_attempted', 'free_throws_made', 'free_throws_attempted', 'offensive_rebounds', 'defensive_rebounds', 'total_rebounds', 'assists', 'steals', 'turnovers', 'blocks_favour', 'blocks_against', 'fouls_committed', 'fouls_received', 'valuation', 'plus_minus'])
df_boxscore_eurocup = df_boxscore_eurocup.drop(columns = ['game_player_id', 'minutes', 'game', 'round', 'phase', 'season_code', 'is_starter', 'is_playing', 'dorsal', 'player', 'two_points_made', 'two_points_attempted', 'three_points_made', 'three_points_attempted', 'free_throws_made', 'free_throws_attempted', 'offensive_rebounds', 'defensive_rebounds', 'total_rebounds', 'assists', 'steals', 'turnovers', 'blocks_favour', 'blocks_against', 'fouls_committed', 'fouls_received', 'valuation', 'plus_minus'])

In [12]:
# Comprobamos que los dataframes ya tienen el formato correcto
df_puntos_euroleague

game_id team_id  points  points_a  points_b
0       E2007_001     BAS       3         0         3
1       E2007_001     OLY       0         0         3
2       E2007_001     OLY       2         2         3
3       E2007_001     BAS       2         2         5
4       E2007_001     OLY       0         2         5
...           ...     ...     ...       ...       ...
642543  E2024_018     ZAL       1        72        68
642544  E2024_018     OLY       1        73        68
642545  E2024_018     OLY       1        74        68
642546  E2024_018     ZAL       0        74        68
642547  E2024_018     ZAL       0        74        68

[642548 rows x 5 columns]

Antes de empezar con los análisis de cada teoría, primero contamos todos los partidos que hay en cada dataframe y los guardaremos en una variable para posteriormente calcular el porcentaje de cumplimiento de cada teoría

In [13]:
total_partidos_euroleague = df_boxscore_euroleague['game_id'].nunique()
total_partidos_eurocup = df_boxscore_eurocup['game_id'].nunique()
total_partidos = total_partidos_euroleague + total_partidos_eurocup
print(total_partidos)

7834


# 1. Teoría del 69

In [14]:
# Empezamos por Euroliga
df_puntos_euroleague['Cumple Teoría 69'] = False
cumple_teoria69_euroleague = 0
for game_id, group in df_puntos_euroleague.groupby('game_id'):
    # Variable para monitorear el primer equipo en alcanzar los 69 puntos
    equipo_69 = None
    
    # Iterar por cada jugada del partido
    for _, row in group.iterrows():
        if row['points_a'] >= 69 or row['points_b'] >= 69:
            equipo_69 = 'A' if row['points_a'] >= 69 else 'B'
            break
            
    # Determinar el equipo ganador según los puntos tras el último tiro del partido
    ganador = 'A' if group.iloc[-1]['points_a'] > group.iloc[-1]['points_b'] else 'B'
    
    # Verificar si el ganador coincide con el equipo que anotó primero
    cumple_teoria = equipo_69 == ganador

    # Actualizar la columna en el DataFrame solo para las filas del partido actual
    df_puntos_euroleague.loc[df_puntos_euroleague['game_id'] == game_id, 'Cumple Teoría 69'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoria69_euroleague += 1


# Cálculo del porcentaje
porcentaje_69_euroleague = cumple_teoria69_euroleague / total_partidos_euroleague * 100
print(f"Porcentaje de éxito de la Teoría del 69 en Euroliga: {porcentaje_69_euroleague:.2f}%")

Porcentaje de éxito de la Teoría del 69 en Euroliga: 82.84%


In [15]:
# Mismo procedimiento para Eurocup
df_puntos_eurocup['Cumple Teoría 69'] = False
cumple_teoria69_eurocup = 0
for game_id, group in df_puntos_eurocup.groupby('game_id'):
    equipo_69 = None
    
    # Iterar por cada jugada del partido
    for _, row in group.iterrows():
        if row['points_a'] >= 69 or row['points_b'] >= 69:
            equipo_69 = 'A' if row['points_a'] >= 69 else 'B'
            break
            
    # Determinar el equipo ganador según los puntos tras el último tiro del partido
    ganador = 'A' if group.iloc[-1]['points_a'] > group.iloc[-1]['points_b'] else 'B'
    
    # Verificar si el ganador coincide con el equipo que anotó primero
    cumple_teoria = equipo_69 == ganador

    # Actualizar la columna en el DataFrame solo para las filas del partido actual
    df_puntos_eurocup.loc[df_puntos_eurocup['game_id'] == game_id, 'Cumple Teoría 69'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoria69_eurocup += 1

# Cálculo del porcentaje
porcentaje_69_eurocup = cumple_teoria69_eurocup / total_partidos_eurocup * 100
print(f"Porcentaje de éxito de la Teoría del 69 en Eurocup: {porcentaje_69_eurocup:.2f}%")

Porcentaje de éxito de la Teoría del 69 en Eurocup: 83.84%


In [16]:
# Calculamos el total entre ambas commpeticiones
cumple_teoria69_total = cumple_teoria69_euroleague + cumple_teoria69_eurocup
porcentaje_69_total = cumple_teoria69_total / total_partidos * 100
print(f"Porcentaje de éxito total de la Teoría del 69: {porcentaje_69_total:.2f}%")

Porcentaje de éxito total de la Teoría del 69: 83.29%


# 2. Teoría del 2-0 inicial

In [17]:
# Empezamos por Euroliga
df_puntos_euroleague['Cumple Teoría 2-0 Inicial'] = False
cumple_teoria20_euroleague = 0
for game_id, group in df_puntos_euroleague.groupby('game_id'):
    # Variables para monitorear los puntos acumulados y el equipo que anota primero
    puntos_a, puntos_b = 0, 0
    equipo_20 = None

    # Iterar por cada jugada del partido
    for _, row in group.iterrows():
        puntos_a += row['points_a']
        puntos_b += row['points_b']

        # Detectar el primer 2-0
        if puntos_a == 2 and puntos_b == 0:
            equipo_20 = 'A'
            break
        elif puntos_b == 2 and puntos_a == 0:
            equipo_20 = 'B'
            break
            
    # Determinar el equipo ganador según los puntos tras el último tiro del partido
    ganador = 'A' if group.iloc[-1]['points_a'] > group.iloc[-1]['points_b'] else 'B'
    
    # Verificar si el ganador coincide con el equipo que anotó primero
    cumple_teoria = equipo_20 == ganador

    # Actualizar la columna en el DataFrame solo para las filas del partido actual
    df_puntos_euroleague.loc[df_puntos_euroleague['game_id'] == game_id, 'Cumple Teoría 2-0 Inicial'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoria20_euroleague += 1

# Cálculo del porcentaje
porcentaje_20_euroleague = cumple_teoria20_euroleague / total_partidos_euroleague * 100
print(f"Porcentaje de éxito de la Teoría del 2-0 Inicial en Euroliga: {porcentaje_20_euroleague:.2f}%")

Porcentaje de éxito de la Teoría del 2-0 Inicial en Euroliga: 39.88%


In [18]:
# Mismo procedimiento para Eurocup
df_puntos_eurocup['Cumple Teoría 2-0 Inicial'] = False
cumple_teoria20_eurocup = 0
for game_id, group in df_puntos_eurocup.groupby('game_id'):
    # Variables para monitorear los puntos acumulados y el equipo que anota primero
    puntos_a, puntos_b = 0, 0
    equipo_20 = None

    # Iterar por cada jugada del partido
    for _, row in group.iterrows():
        puntos_a += row['points_a']
        puntos_b += row['points_b']

        # Detectar el primer 2-0
        if puntos_a == 2 and puntos_b == 0:
            equipo_20 = 'A'
            break
        elif puntos_b == 2 and puntos_a == 0:
            equipo_20 = 'B'
            break
            
    # Determinar el equipo ganador tras los puntos en el último tiro del partido
    ganador = 'A' if group.iloc[-1]['points_a'] > group.iloc[-1]['points_b'] else 'B'
    
    # Verificar si el ganador coincide con el equipo que anotó primero
    cumple_teoria = equipo_20 == ganador

    # Actualizar la columna en el DataFrame solo para las filas del partido actual
    df_puntos_eurocup.loc[df_puntos_eurocup['game_id'] == game_id, 'Cumple Teoría 2-0 Inicial'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoria20_eurocup += 1

# Cálculo del porcentaje
porcentaje_20_eurocup = cumple_teoria20_eurocup / total_partidos_eurocup * 100
print(f"Porcentaje de éxito de la Teoría del 2-0 Inicial en Eurocup: {porcentaje_20_eurocup:.2f}%")

Porcentaje de éxito de la Teoría del 2-0 Inicial en Eurocup: 40.41%


In [19]:
# Calculamos el total entre ambas commpeticiones
cumple_teoria20_total = cumple_teoria20_euroleague + cumple_teoria20_eurocup
porcentaje_20_total = cumple_teoria20_total / total_partidos * 100
print(f"Porcentaje de éxito total de la Teoría del 2-0 Inicial: {porcentaje_20_total:.2f}%")

Porcentaje de éxito total de la Teoría del 2-0 Inicial: 40.12%


# 3. Análisis de la Teoría de los Dobles Dígitos

In [20]:
# Empezamos por Euroliga
df_puntos_euroleague['Cumple Teoría Dobles Dígitos'] = False
cumple_teoria10_euroleague = 0
for game_id, group in df_puntos_euroleague.groupby('game_id'):
    # Variables para monitorear el equipo que llega primero a 10
    equipo_10 = None
    
    # Iterar por cada jugada del partido
    for _, row in group.iterrows():
        if row['points_a'] >= 10 or row['points_b'] >= 10:
            equipo_10 = 'A' if row['points_a'] >= 10 else 'B'
            break
            
    # Determinar el equipo ganador tras los puntos en el último tiro del partido
    ganador = 'A' if group.iloc[-1]['points_a'] > group.iloc[-1]['points_b'] else 'B'
    
    # Verificar si el ganador coincide con el equipo que llegó primero a 10 puntos
    cumple_teoria = equipo_10 == ganador

    # Actualizar la columna en el DataFrame solo para las filas del partido actual
    df_puntos_euroleague.loc[df_puntos_euroleague['game_id'] == game_id, 'Cumple Teoría Dobles Dígitos'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoria10_euroleague += 1


# Cálculo del porcentaje
porcentaje_10_euroleague = cumple_teoria10_euroleague / total_partidos_euroleague * 100
print(f"Porcentaje de éxito de la Teoría de los Dobles Dígitos en Euroliga: {porcentaje_10_euroleague:.2f}%")

Porcentaje de éxito de la Teoría de los Dobles Dígitos en Euroliga: 61.73%


In [21]:
# Mismo procedimiento para Eurocup
df_puntos_eurocup['Cumple Teoría Dobles Dígitos'] = False
cumple_teoria10_eurocup = 0
for game_id, group in df_puntos_eurocup.groupby('game_id'):
    # Variables para monitorear el equipo que llega primero a 10
    equipo_10 = None
    
    # Iterar por cada jugada del partido
    for _, row in group.iterrows():
        if row['points_a'] >= 10 or row['points_b'] >= 10:
            equipo_10 = 'A' if row['points_a'] >= 10 else 'B'
            break
            
    # Determinar el equipo ganador tras los puntos en el último tiro del partido
    ganador = 'A' if group.iloc[-1]['points_a'] > group.iloc[-1]['points_b'] else 'B'
    
    # Verificar si el ganador coincide con el equipo que llegó primero a 10 puntos
    cumple_teoria = equipo_10 == ganador

    # Actualizar la columna en el DataFrame solo para las filas del partido actual
    df_puntos_eurocup.loc[df_puntos_eurocup['game_id'] == game_id, 'Cumple Teoría Dobles Dígitos'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoria10_eurocup += 1


# Cálculo del porcentaje
porcentaje_10_eurocup = cumple_teoria10_eurocup / total_partidos_eurocup * 100
print(f"Porcentaje de éxito de la Teoría de los Dobles Dígitos en Eurocup: {porcentaje_10_eurocup:.2f}%")

Porcentaje de éxito de la Teoría de los Dobles Dígitos en Eurocup: 62.18%


In [22]:
# Calculamos el total entre ambas commpeticiones
cumple_teoria10_total = cumple_teoria10_euroleague + cumple_teoria10_eurocup
porcentaje_10_total = cumple_teoria10_total / total_partidos * 100
print(f"Porcentaje de éxito total de la Teoría de los Dobles Dígitos: {porcentaje_10_total:.2f}%")

Porcentaje de éxito total de la Teoría de los Dobles Dígitos: 61.94%


# 4. Análisis de la Teoría de Jugadores con 10+ Puntos

Antes de comenzar con el análisis de la teoría, es necesario realizar algunos pasos para preparar los datos:

1. Asignar los equipos team_a y team_b en el dataframe de puntos (df_puntos) según los cambios en el marcador (points_a y points_b).
2. Vincular los identificadores reales de los equipos (team_id) con team_a y team_b para identificarlos claramente.
3. Combinar el dataframe de puntos con el boxscore (df_boxscore) para relacionar las estadísticas de jugadores con la información del equipo y el resultado del partido.

In [23]:
# Crear un DataFrame para almacenar los equipos asociados a cada partido
equipos_asociados = {}

# Iterar por cada partido
for game_id, group in df_puntos_euroleague.groupby('game_id'):
    team_a, team_b = None, None
    prev_points_a, prev_points_b = 0, 0  # Puntos iniciales
    for _, row in group.iterrows():
        # Si cambia points_a, asociamos ese team_id a team_a
        if row['points_a'] != prev_points_a and team_a is None:
            team_a = row['team_id']
        # Si cambia points_b, asociamos ese team_id a team_b
        if row['points_b'] != prev_points_b and team_b is None:
            team_b = row['team_id']

        # Actualizamos los valores previos
        prev_points_a = row['points_a']
        prev_points_b = row['points_b']

        # Si ya tenemos ambos equipos, podemos detenernos
        if team_a and team_b:
            break

    # Guardamos los equipos para el partido actual
    equipos_asociados[game_id] = {'team_a': team_a, 'team_b': team_b}

# Convertir el diccionario en un DataFrame
equipos_df = pd.DataFrame.from_dict(equipos_asociados, orient='index').reset_index()
equipos_df.columns = ['game_id', 'team_a', 'team_b']

# Unir con el marcador final
df_resultados_euroleague = df_puntos_euroleague.groupby('game_id').tail(1).merge(
    equipos_df, on='game_id'
)

# Agregar los puntos finales de cada equipo
df_resultados_euroleague['points_team_a'] = df_resultados_euroleague['points_a']
df_resultados_euroleague['points_team_b'] = df_resultados_euroleague['points_b']


df_resultados_euroleague.tail()
        

game_id team_id  points  points_a  points_b  Cumple Teoría 69  \
4320  E2024_014     PAN       2        94        79              True   
4321  E2024_015     PAR       0        93        86              True   
4322  E2024_016     PRS       0        79        74             False   
4323  E2024_017     BAR       0        88        73              True   
4324  E2024_018     ZAL       0        74        68              True   

      Cumple Teoría 2-0 Inicial  Cumple Teoría Dobles Dígitos team_a team_b  \
4320                       True                          True    PAN    MUN   
4321                      False                         False    MAD    PAR   
4322                      False                          True    MIL    PRS   
4323                       True                          True    BAR    BER   
4324                       True                          True    OLY    ZAL   

      points_team_a  points_team_b  
4320             94             79  
4321             93             86  
4322             79             74  
4323             88             73  
4324             74             68

In [24]:
# Juntar el dataframe resultante con el boxscore
df_conjunto_euroleague = df_boxscore_euroleague.merge(
    df_resultados_euroleague[['game_id', 'team_a', 'team_b', 'points_team_a', 'points_team_b']],
    on='game_id'
)
df_conjunto_euroleague.head()

game_id player_id team_id  points team_a team_b  points_team_a  \
0  E2007_191      PADS     BAR      19    BAR    ROM             86   
1  E2007_191      PAZU     BAR       3    BAR    ROM             86   
2  E2007_191      PJJF     BAR       8    BAR    ROM             86   
3  E2007_191      PAHW     BAR       4    BAR    ROM             86   
4  E2007_191      PATA     BAR      17    BAR    ROM             86   

   points_team_b  
0             57  
1             57  
2             57  
3             57  
4             57

Repetimos el mismo proceso para la Eurocup.

In [25]:
# Crear un DataFrame para almacenar los equipos asociados a cada partido
equipos_asociados = {}

# Iterar por cada partido
for game_id, group in df_puntos_eurocup.groupby('game_id'):
    team_a, team_b = None, None
    prev_points_a, prev_points_b = 0, 0  # Puntos iniciales
    for _, row in group.iterrows():
        # Si cambia points_a, asociamos ese team_id a team_a
        if row['points_a'] != prev_points_a and team_a is None:
            team_a = row['team_id']
        # Si cambia points_b, asociamos ese team_id a team_b
        if row['points_b'] != prev_points_b and team_b is None:
            team_b = row['team_id']

        # Actualizamos los valores previos
        prev_points_a = row['points_a']
        prev_points_b = row['points_b']

        # Si ya tenemos ambos equipos, podemos detenernos
        if team_a and team_b:
            break

    # Guardamos los equipos para el partido actual
    equipos_asociados[game_id] = {'team_a': team_a, 'team_b': team_b}

# Convertir el diccionario en un DataFrame
equipos_df = pd.DataFrame.from_dict(equipos_asociados, orient='index').reset_index()
equipos_df.columns = ['game_id', 'team_a', 'team_b']

# Unir con el marcador final
df_resultados_eurocup = df_puntos_eurocup.groupby('game_id').tail(1).merge(
    equipos_df, on='game_id'
)

# Agregar los puntos finales de cada equipo
df_resultados_eurocup['points_team_a'] = df_resultados_eurocup['points_a']
df_resultados_eurocup['points_team_b'] = df_resultados_eurocup['points_b']


df_resultados_eurocup.tail()

game_id team_id  points  points_a  points_b  Cumple Teoría 69  \
3504  U2023_191     LLI       2        85        93              True   
3505  U2023_192     BES       1        78        69              True   
3506  U2023_194     BOU       0        89        63              True   
3507  U2023_195     BOU       2        77        64              True   
3508  U2023_196     BOU       2        81        89              True   

      Cumple Teoría 2-0 Inicial  Cumple Teoría Dobles Dígitos team_a team_b  \
3504                       True                          True    LLI    PRS   
3505                      False                          True    BES    BOU   
3506                       True                          True    BOU    BES   
3507                      False                          True    PRS    BOU   
3508                      False                         False    BOU    PRS   

      points_team_a  points_team_b  
3504             85             93  
3505             78             69  
3506             89             63  
3507             77             64  
3508             81             89

In [26]:
# Juntar el dataframe resultante con el boxscore
df_conjunto_eurocup = df_boxscore_eurocup.merge(
    df_resultados_eurocup[['game_id', 'team_a', 'team_b', 'points_team_a', 'points_team_b']],
    on='game_id'
)
df_conjunto_eurocup.head()

game_id player_id team_id  points team_a team_b  points_team_a  \
0  U2007_187      PJEZ     JER       0    JER    DEN            112   
1  U2007_187      PKMF     JER       3    JER    DEN            112   
2  U2007_187      PKNU     JER      23    JER    DEN            112   
3  U2007_187      PKQK     JER       7    JER    DEN            112   
4  U2007_187   P000678     JER      21    JER    DEN            112   

   points_team_b  
0             97  
1             97  
2             97  
3             97  
4             97

Tras hacerlo para ambas competiciones y tener el dataframe preparado, vamos con la teoría

In [27]:
# Inicializamos el contador para la teoría
df_conjunto_euroleague['Cumple Teoría Jugadores'] = False
cumple_teoriajugadores_euroleague = 0

# Iteramos por cada partido
for game_id, group in df_conjunto_euroleague.groupby('game_id'):
     # Contadores para jugadores que alcanzan 10 puntos por equipo
    jugadores_a, jugadores_b = 0, 0
    
    # Contamos cuántos jugadores alcanzaron 10 puntos o más en cada equipo
    for _, row in group.iterrows():
        if row['team_id'] == group['team_a'].iloc[0] and row['points'] >= 10:
            jugadores_a += 1
        elif row['team_id'] == group['team_b'].iloc[0] and row['points'] >= 10:
            jugadores_b += 1

    # Determinar el equipo ganador
    ganador = 'A' if group['points_team_a'].iloc[0] > group['points_team_b'].iloc[0] else 'B'

    # Verificar si el equipo con más jugadores con 10+ puntos ganó el partido
    cumple_teoria = (jugadores_a > jugadores_b and ganador == 'A') or (jugadores_b > jugadores_a and ganador == 'B')

    # Actualizar la columna para las filas del partido actual
    df_conjunto_euroleague.loc[df_conjunto_euroleague['game_id'] == game_id, 'Cumple Teoría Jugadores'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoriajugadores_euroleague += 1

# Calcular el porcentaje de éxito de la teoría
porcentaje_jugadores_euroleague = (cumple_teoriajugadores_euroleague / total_partidos_euroleague) * 100
print(f"Porcentaje de éxito de la Teoría de Jugadores con 10+ Puntos en Euroliga: {porcentaje_jugadores_euroleague:.2f}%")

Porcentaje de éxito de la Teoría de Jugadores con 10+ Puntos en Euroliga: 55.82%


In [28]:
# Inicializamos el contador para la teoría
df_conjunto_eurocup['Cumple Teoría Jugadores'] = False
cumple_teoriajugadores_eurocup = 0

# Iteramos por cada partido
for game_id, group in df_conjunto_eurocup.groupby('game_id'):
     # Contadores para jugadores que alcanzan 10 puntos por equipo
    jugadores_a, jugadores_b = 0, 0
    
    # Contamos cuántos jugadores alcanzaron 10 puntos o más en cada equipo
    for _, row in group.iterrows():
        if row['team_id'] == group['team_a'].iloc[0] and row['points'] >= 10:
            jugadores_a += 1
        elif row['team_id'] == group['team_b'].iloc[0] and row['points'] >= 10:
            jugadores_b += 1

    # Determinar el equipo ganador
    ganador = 'A' if group['points_team_a'].iloc[0] > group['points_team_b'].iloc[0] else 'B'

    # Verificar si el equipo con más jugadores con 10+ puntos ganó el partido
    cumple_teoria = (jugadores_a > jugadores_b and ganador == 'A') or (jugadores_b > jugadores_a and ganador == 'B')

    # Actualizar la columna para las filas del partido actual
    df_conjunto_eurocup.loc[df_conjunto_eurocup['game_id'] == game_id, 'Cumple Teoría Jugadores'] = cumple_teoria

    # Incrementar el contador si la teoría se cumple
    if cumple_teoria:
        cumple_teoriajugadores_eurocup += 1

# Calcular el porcentaje de éxito de la teoría
porcentaje_jugadores_eurocup = (cumple_teoriajugadores_eurocup / total_partidos_eurocup) * 100
print(f"Porcentaje de éxito de la Teoría de Jugadores con 10+ Puntos en Eurocup: {porcentaje_jugadores_eurocup:.2f}%")

Porcentaje de éxito de la Teoría de Jugadores con 10+ Puntos en Eurocup: 55.46%


In [29]:
cumple_teoriajugadores_total = cumple_teoriajugadores_euroleague + cumple_teoriajugadores_eurocup
porcentaje_jugadores_total = cumple_teoriajugadores_total / total_partidos * 100
print(f"Porcentaje de éxito total de la Teoría de de Jugadores con 10+ Puntos: {porcentaje_jugadores_total:.2f}%")

Porcentaje de éxito total de la Teoría de de Jugadores con 10+ Puntos: 55.65%


Por orden de cumplimiento:
1. Teoría del 69: 83.29%
2. Teoría de los Dobles Dígitos Primero: 61.94%
3. Teoría de Jugadores con 10+ Puntos: 55.65%
4. Teoría del 2-0 Inicial: 40.12%

In [30]:
# Guardo los dataframes actualizados en formato Excel para importarlos en Tableau y crear las visualizaciones (ajustar la ruta según tus directorios)
df_puntos_euroleague.to_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\puntos_euroleague_cleaned.csv", index=False)
df_puntos_eurocup.to_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\puntos_eurocup_cleaned.csv", index=False)

df_conjunto_euroleague.to_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\jugadores_euroleague_cleaned.csv", index=False)
df_conjunto_eurocup.to_csv(r"C:\Users\Yago\Documents\Basketmática\Teoria69_Noviembre2024\jugadores_eurocup_cleaned.csv", index=False)